api-controllers/ServiceFabrikAdminController.js

```diff
@@ -243,14 +243,16 @@ class ServiceFabrikAdminController extends FabrikBaseController {
 
   getDeployment(req, res) {
     const deploymentName = req.params.name;
+    const plan = catalog.getPlan(req.query.plan_id);
     this.createManager(req.query.plan_id)
-      .then(manager => this.cloudController.getOrgAndSpaceGuid(this.getInstanceId(deploymentName))
-        .then(opts => {
-          const context = {
-            platform: CONST.PLATFORM.CF,
-            organization_guid: opts.organization_guid,
-            space_guid: opts.space_guid
-          };
+      .then(manager =>
+        eventmesh.apiServerClient.getPlatformContext({
```

Comment:
"Have we stored all deployments" => Assuming that all the active deployments have corresponding resource. As if it was not the case, it would have created issues at lot of other places also (ex. backup)

"do all deployment resource have the context?" => Except for deployments for which delete failed, won't have context as earlier we were not merging options field. I checked in canary, all the instances for which delete failed were not active instances hence assumed here that all the active instances will have platform context.
After discussing with @ketakig12  @subhankarc decided to go ahead with assumption that context will always be there in resource

Reply:
Additionally we can throw InternalServerError if context is undefined from ApiServerClient